### ISSUE TITLE
OS X: read /etc/manpaths and /etc/manpaths.d

### ISSUE THREAD

**Issue by daenney** [fish-shell/fish-shell#1092]
Similar to `/etc/paths` and `/etc/paths.d` OS X has `/etc/manpaths` and `/etc/manpaths.d` which should be used to populate the `$MANPATH` environment variable.

**Comment by kballard** [comment]
OS X uses a tool `/usr/libexec/path_helper` to construct the default `PATH` and `MANPATH` vars. This tool only defines `MANPATH` if the `MANPATH` variable is already set. Since it's executed as the first thing in `/etc/profile`, which is the first file Bash sources, AFAIK the only way to actually have the `MANPATH` be set up is if Bash inherits a `MANPATH` variable from its environment.

To that end, if we do adopt this same behavior (and we don't decide to do it by invoking `path_helper` ourselves and parsing the output into something Fish-compatible), we should match that expectation and only set `MANPATH` if it's already set. The reason for this behavior is because, by leaving `MANPATH` un-set, `man` constructs its own search path based on the `PATH` variable.

**Comment by krader1961** [comment]
@kballard I wrote this [blog article](http://www.skepticism.us/2015/10/mac-os-x-man-command-ignores-manpath-which-sucks-for-homebrew-installed-commands/) about why MANPATH doesn't work on OS X less than six months ago. I admit I did not expend much effort in figuring out why it didn't work.  But I eventually gave up and added the following comment to my *environ.fish* file (and before that the equivalent zsh file) which is sourced from my *~/.config/fish/config.fish* file:

```
# Setting $MANPATH is pointless on Mac OS X as it is ignored. Only
# paths in /etc/man.conf are recognized.
```

I would not be offended to learn that despite using OS X for over a decade I misunderstood this aspect of its behavior.

**Comment by kballard** [comment]
@krader1961 That blog post is rather perplexing, because I just tried and it worked just fine.

```
> brew install coreutils
[...]
> env MANPATH=/usr/local/opt/coreutils/libexec/gnuman man -w ls
/usr/local/opt/coreutils/libexec/gnuman/man1/ls.1
```

If I run a Bash login shell with that `MANPATH` set, so as to make `/usr/libexec/path_helper` give me a complete `MANPATH` that includes the built-in stuff, then it sorts it such that `man ls` still gives me the built-in one, but `man -a ls` gives me both the built-in and the coreutils manpage.

**Comment by daenney** [comment]
I've so far never had any issues with OS X ignoring or not respecting MANPATH. Setting it is how I make sure all the man pages that get coupled with projects installed from brew show up when I do a `man <thing from brew>`.

**Comment by krader1961** [comment]
This has been resolved but wasn't automatically closed because the commit did not include the magic line "Fixes #1092".
